src/flockwave/queries.js

```diff
@@ -14,6 +14,40 @@ import { errorToString } from '~/error-handling';
 import { extractResponseForId } from './parsing';
 import { validateExtensionName } from './validation';
 
+/**
+ * Adapts the given base64-encoded show using the given transformation
+ * definitions and coordinate system.
+ */
+export async function adaptShow(hub, show, transformations, coordinateSystem) {
+  const response = await hub.sendMessage(
+    {
+      type: 'X-SHOW-ADAPT',
+      show,
+      transformations,
+      environment: {
+        location: {
+          origin: [
+            // Convert the origin to a lat-lon integer pair.
+            // Unit must be 1e-7 degrees (so multiple by 1e7).
+            Math.round(coordinateSystem.origin[1] * 1e7),
```

Comment:
We have a `toScaledJSONFromLonLat()` function for this type of conversion so I will just use that here.